src/app/components/(imageAnimation)/page.tsx

```diff
@@ -0,0 +1,447 @@
+  },
+  {
+    id: 2,
+    title: 'Immersive Experience Portal',
+    description:
+      'Cutting-edge visualization technologies combined with intuitive navigation for creating memorable digital journeys.',
+    imageSrc: '/images/person/person2.webp'
+  },
+  {
+    id: 3,
+    title: 'Advanced Analytics Dashboard',
+    description:
+      'Real-time data processing and visualization tools that provide actionable insights through elegant and accessible interfaces.',
+    imageSrc: '/images/person/person3.webp'
+  }
+];
+
+// 애니메이션 텍스트 컴포넌트
+const AnimatedLetters = ({ text, delay = 0, className = '' }: { text: string; delay?: number; className?: string }) => {
+  return (
+    <motion.div className={`flex flex-wrap items-center justify-center ${className}`}>
+      {text.split('').map((char: string, i: number) => (
+        <motion.span
+          key={i}
+          className="inline-block"
+          initial={{ opacity: 0, y: 20, scale: 0.8 }}
+          whileInView={{ opacity: 1, y: 0, scale: 1 }}
+          transition={{
+            duration: 0.6,
+            delay: delay + i * 0.04,
+            type: 'spring',
+            stiffness: 100,
+            damping: 10
+          }}
+          viewport={{ once: false, amount: 0.1 }}
+        >
+          {char === ' ' ? '\u00A0' : char}
+        </motion.span>
+      ))}
+    </motion.div>
+  );
+};
+
+// 애니메이션 단어 컴포넌트
+const AnimatedWords = ({ text, delay = 0, className = '' }: { text: string; delay?: number; className?: string }) => {
+  return (
+    <motion.div className={`flex flex-wrap items-center justify-center ${className}`}>
+      {text.split(' ').map((word: string, i: number) => (
+        <motion.span
+          key={i}
+          className="mr-2 inline-block"
+          initial={{ opacity: 0, y: i % 2 === 0 ? 20 : -20, x: i % 3 === 0 ? -10 : i % 3 === 1 ? 10 : 0 }}
+          whileInView={{ opacity: 1, y: 0, x: 0 }}
+          transition={{
+            duration: 0.7,
+            delay: delay + i * 0.07,
+            type: 'spring',
+            stiffness: 100
+          }}
+          viewport={{ once: true, amount: 1 }}
+        >
+          {word}
+        </motion.span>
+      ))}
+    </motion.div>
+  );
+};
+
+// 스크롤 기반 애니메이션 값 추출 함수
+const useScrollAnimation = (
+  scrollYProgress: MotionValue<number>,
+  outputRange: Array<number | string>,
+  inputRange: number[] = [0, 1]
+) => {
+  return useTransform(scrollYProgress, inputRange, outputRange);
+};
+
+export default function ImageAnimation() {
+  const containerRef = useRef(null);
+  const [activeSection, setActiveSection] = useState(0);
+
+  // 전체 스크롤 진행률
+  const { scrollYProgress } = useScroll({
+    target: containerRef,
+    offset: ['start start', 'end end']
+  });
+
+  // 각 섹션의 시작과 끝 범위 정의 (0-1 사이의 값)
+  // 섹션 간에 겹치는 부분 없이 연속적으로 배치
+  const sectionRanges = [
+    [0.05, 0.35], // 첫 번째 섹션 (5%~35%)
+    [0.35, 0.65], // 두 번째 섹션 (35%~65%)
+    [0.7, 0.95] // 세 번째 섹션 (70%~95%)
+  ];
```

Comment:
아래 `액티브 섹션 추적` useEffect에 의존성 배열 경고가 떠서 `useMemo()`로 처리했어요!

```suggestion
  const sectionRanges = useMemo(() => [
    [0.05, 0.35], // 첫 번째 섹션 (5%~35%)
    [0.35, 0.65], // 두 번째 섹션 (35%~65%)
    [0.7, 0.95] // 세 번째 섹션 (70%~95%)
  ], []);
```